File: Tutorial_Sample_Code/ViewerObjectModel/CustomViewer/VB/VB_Win_CRVObjMod_CustomizeViewer/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared
Imports CrystalDecisions.Windows.Forms

Public Class Form1

    Private Sub ConfigureCrystalReports()
        listCRVReport.DataSource = System.Enum.GetValues(GetType(CeWinCRVReportOptions))
        listCRVToolbar.DataSource = System.Enum.GetValues(GetType(CeWinCRVToolbarOptions))
        myCrystalReportViewer.ReportSource = "C:\WorldSalesReport.rpt"

        selectBackColor.DataSource = System.Enum.GetValues(GetType(KnownColor))
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub

    Private Sub redisplay_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles redisplay.Click
        myCrystalReportViewer.ShowPageNavigateButtons = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Page_Navigation_Button)
        myCrystalReportViewer.ShowGotoPageButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Go_to_Page_Button)
        myCrystalReportViewer.ShowCloseButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Close_View_Button)
        myCrystalReportViewer.ShowPrintButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Print_Button)
        myCrystalReportViewer.ShowRefreshButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Refresh_Button)
        myCrystalReportViewer.ShowExportButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Export_Button)
        myCrystalReportViewer.ShowGroupTreeButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Group_Tree_Button)
        myCrystalReportViewer.ShowZoomButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Zoom_Button)
        myCrystalReportViewer.ShowTextSearchButton = listCRVToolbar.GetSelected(CeWinCRVToolbarOptions.Search_Button)

        myCrystalReportViewer.DisplayToolbar = listCRVReport.GetSelected(CeWinCRVReportOptions.Toolbar)

        If (listCRVReport.GetSelected(Convert.ToInt32(CeWinCRVReportOptions.Group_Tree))) Then
            myCrystalReportViewer.ToolPanelView = ToolPanelViewType.GroupTree
        Else
            myCrystalReportViewer.ToolPanelView = ToolPanelViewType.None
        End If

        myCrystalReportViewer.DisplayStatusBar = listCRVReport.GetSelected(CeWinCRVReportOptions.Status_Bar)

        Dim mySelectedKnownColor As KnownColor = CType(selectBackColor.SelectedItem, KnownColor)
        If Not mySelectedKnownColor = KnownColor.Transparent Then
            myCrystalReportViewer.BackColor = System.Drawing.Color.FromKnownColor(mySelectedKnownColor)
        End If

    End Sub

    Private Sub goToPage_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles goToPage.Click
        myCrystalReportViewer.ShowNthPage(Convert.ToInt32(pageNumber.Text))

    End Sub

    Private Sub updateZoomFactor_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles updateZoomFactor.Click
        myCrystalReportViewer.Zoom(Convert.ToInt32(zoomFactor.Text))
    End Sub

    Private Sub search_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles search.Click
        Dim mySearchResult As Boolean = myCrystalReportViewer.SearchForText(searchText.Text)
        If mySearchResult Then
            message.Text = MessageConstants.SUCCESS
        Else
            message.Text = MessageConstants.FAILURE
        End If
    End Sub
End Class

File: Tutorial_Sample_Code/ViewerObjectModel/CustomViewer/CS/CS_Win_CRVObjMod_CustomizeViewer/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Shared;
using CrystalDecisions.Windows.Forms;

namespace CS_Win_CRVObjMod_CustomizeViewer
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void ConfigureCrystalReports()
        {
            listCRVReport.DataSource = System.Enum.GetValues(typeof(CeWinCRVReportOptions));
            listCRVToolbar.DataSource = System.Enum.GetValues(typeof(CeWinCRVToolbarOptions));
            crystalReportViewer.ReportSource = "C:\\WorldSalesReport.rpt";

            selectBackColor.DataSource = System.Enum.GetValues(typeof(KnownColor));

        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void crystalReportViewer_Load(object sender, EventArgs e)
        {

        }
    
        private void redisplay_Click(object sender, EventArgs e)
        {
            crystalReportViewer.ShowPageNavigateButtons = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Page_Navigation_Button));
            crystalReportViewer.ShowGotoPageButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Go_to_Page_Button));
            crystalReportViewer.ShowCloseButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Close_View_Button));
            crystalReportViewer.ShowPrintButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Print_Button));
            crystalReportViewer.ShowRefreshButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Refresh_Button));
            crystalReportViewer.ShowExportButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Export_Button));
            crystalReportViewer.ShowGroupTreeButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Group_Tree_Button));
            crystalReportViewer.ShowZoomButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Zoom_Button));
            crystalReportViewer.ShowTextSearchButton = listCRVToolbar.GetSelected(Convert.ToInt32(CeWinCRVToolbarOptions.Search_Button));

            crystalReportViewer.DisplayToolbar = listCRVReport.GetSelected(Convert.ToInt32(CeWinCRVReportOptions.Toolbar));
            if (listCRVReport.GetSelected(Convert.ToInt32(CeWinCRVReportOptions.Group_Tree)))
            {
                crystalReportViewer.ToolPanelView = ToolPanelViewType.GroupTree;
            }
            else
            {
                crystalReportViewer.ToolPanelView = ToolPanelViewType.None;
            }
            crystalReportViewer.DisplayStatusBar = listCRVReport.GetSelected(Convert.ToInt32(CeWinCRVReportOptions.Status_Bar));

            KnownColor selectedKnownColor = (KnownColor)selectBackColor.SelectedItem;
            if (selectedKnownColor != KnownColor.Transparent)
            {
                crystalReportViewer.BackColor = System.Drawing.Color.FromKnownColor(selectedKnownColor);
            }
        }

        private void goToPage_Click(object sender, EventArgs e)
        {
            crystalReportViewer.ShowNthPage(Convert.ToInt32(pageNumber.Text));
        }

        private void updateZoomFactor_Click(object sender, EventArgs e)
        {
            crystalReportViewer.Zoom(Convert.ToInt32(zoomFactor.Text));
        }

        private void search_Click(object sender, EventArgs e)
        {
            bool searchResult = crystalReportViewer.SearchForText(searchText.Text);
            if (searchResult)
            {
                message.Text = MessageConstants.SUCCESS;
            }
            else
            {
                message.Text = MessageConstants.FAILURE;
            }

        }
    }
}